src/ui/interfaces/option-select-config.ts

```diff
@@ -0,0 +1,80 @@
+// -- start tsdoc imports --
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { Mode } from "#app/ui/ui";
+// -- end tsdoc imports --
+import type { TextStyle } from "#app/ui/text";
+
+/**
+ * Customizations options for UI's {@linkcode Mode.OPTION_SELECT}
+ *
+ * @template T the specifc type of {@linkcode OptionSelectItem} contained by this config
+ */
+export interface OptionSelectModeConfig<T extends OptionSelectItem> extends OptionMenuSettings {
```

Comment:
NIT: Considering that in 90% of the cases it's being used like this:
```ts
const config: OptionSelectModeConfig<OptionSelectItem> = // [...]
```

It might be worth to set this as default/omitting it in that case:

```suggestion
export interface OptionSelectModeConfig<T extends OptionSelectItem = OptionSelectItem> extends OptionMenuSettings {
```

---->

```ts
const config: OptionSelectModeConfig = // [...]
```

Reply:
ooh, didn't know you could do that, thanks! Having to add it every time was annoying and I almost considered defining a type just to avoid it, this is much better. Done!